discretisedfield/tests/test_field.py

```diff
@@ -2201,3 +2200,204 @@ def test_numpy_ufunc(self):
+                    assert np.array_equal(
+                        np.array(list(f.mesh.axis_points(i))),
+                        fxa[i].values
+                    )
+                    assert fxa[i].attrs['units'] == f.mesh.attributes['unit']
+                assert 'comp' not in fxa.dims
+                assert np.array_equal(f.array.squeeze(axis=-1), fxa.values)
+
+        f6d = self.pf << self.pf
+        f6d_xa = f6d.to_xarray()
+        assert f6d_xa['comp'].size == 6
+        assert 'comp' not in f6d_xa.coords
+
+        # test name and units defaults
+        f3d_xa = self.pf.to_xarray()
+        assert f3d_xa.name == 'field'
+        assert f3d_xa.attrs['units'] is None
+
+        # test name and units
+        f3d_xa_2 = self.pf.to_xarray(name='m', units='A/m')
+        assert f3d_xa_2.name == 'm'
+        assert f3d_xa_2.attrs['units'] == 'A/m'
+
+    def test_to_xarray_invalid_args(self):
+        args = [
+            ['m', 42.0], [21.0, 42], [21, 'A/m'],
+            [{'name': 'm'}, {'units': 'A/m'}], [['m'], ['A/m']],
+            [['m', 'A/m'], None], [('m', 'A/m'), None],
+            [{'name': 'm', 'units': 'A/m'}, None]
+        ]
+
+        for name, units in args:
+            with pytest.raises(TypeError):
+                fxa = self.pf.to_xarray(name, units)
+
+    def test_from_xarray_valid_args(self):
+        for mesh in self.meshes:
+            for value, dtype in self.vfuncs:
+                f = df.Field(mesh, dim=3, value=value, dtype=dtype)
+                fxa = f.to_xarray()
+                f_new = df.Field.from_xarray(fxa)
+                assert f_new == f  # or use allclose()
+
+            for value, dtype in self.sfuncs:
+                f = df.Field(mesh, dim=1, value=value, dtype=dtype)
+                fxa = f.to_xarray()
+                f_new = df.Field.from_xarray(fxa)
+                assert f_new == f  # or use allclose()
+
+        f_plane = self.pf.plane('z')
+        f_plane_xa = f_plane.to_xarray()
+        f_plane_new = df.Field.from_xarray(f_plane_xa)
+        assert f_plane_new == f_plane  # or use allclose()
+
+        f6d = self.pf << self.pf
+        f6d_xa = f6d.to_xarray()
+        f6d_new = df.Field.from_xarray(f6d_xa)
+        assert f6d_new == f6d  # or use allclose()
+
+    def test_from_xarray_invalid_args_and_DataArrays(self):
+        args = [int(), float(), str(), list(), dict(), xr.Dataset(),
+                np.empty((20, 20, 20, 3))]
+        bad_value1 = xr.DataArray(np.ones((20, 20, 5, 3), dtype=np.int8),
+                                  dims=['x', 'y', 'z', 'comp'],
+                                  coords=dict(x=np.arange(0, 20),
+                                              y=np.arange(0, 20),
+                                              z=np.arange(0, 5),
+                                              comp=['x', 'y', 'z']),
+                                  name='mag',
+                                  attrs=dict(units='A/m'))
```

Comment:
That should be valid in my opinion.

Reply:
Yes, I agree. We should include integer data types.